gem_prelude.rb

```diff
@@ -1,2 +1,11 @@
-require 'rubygems.rb' if defined?(Gem)
-require 'did_you_mean' if defined?(DidYouMean)
+begin
+  require 'rubygems.rb'
+rescue LoadError
+  warn "`RubyGems' were not loaded."
+end if defined?(Gem)
+
+begin
+  require 'did_you_mean'
+rescue Gem::LoadError, LoadError
```

Comment:
`Gem::LoadError` seems wrong here, for the case of `--disable-gems --enable-did_you_mean`

Reply:
Good point. I'll take a look

Reply:
Just `rescue LoadError` should suffice, since `Gem::LoadError` inherits from `LoadError`:

https://github.com/ruby/ruby/blob/30dbd9e4c7e29e4489a8729bfcd2e21e004f7ad9/lib/rubygems/errors.rb#L15

```suggestion
rescue LoadError
```

Reply:
Should be resolved now.